### PR TITLE
C104 rest fullapi

### PR DESCRIPTION
**Pull request by raj8789**
In This branch i have added RetrieveTweets.java class used for retrieving tweeter timeline from tweeter account.
SendTweet.java class file is for sending tweet at the user timeline .
TwitterWorks.java contains method Sendtweet() for posting Tweet on Timeline.
method timeline() to fetch tweets from tweeter account.

### CODE REVIEW

**Comment by raj8789** [comment]
review the code

File: pom.xml
Comment (PoojaMohite-Khoros) [inline pom.xml]: twitter4j-core
dependency have been added twice, please add only 1 , which every latest u want to use

File: src/main/java/com/resource/RetrieveTweets.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/RetrieveTweets.java]: Remove all system out from the PR

File: src/main/java/com/resource/SendTweet.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/SendTweet.java]: Clean this comments, unnecessary code

File: src/main/java/com/resource/TwitterWorks.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/TwitterWorks.java]: here, unnecessary code , comments should be cleaned

File: src/main/java/com/resource/TwitterWorks.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/TwitterWorks.java]: Give a sensibel path
@Path("/getTweet") or /getTimeline

File: src/main/java/com/resource/TwitterWorks.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/TwitterWorks.java]: POST, take class a parameter and not String
class TweetPostRequest {
String message;
// constructor
TweetPostRequest(){
}
// getter
public String getMessage(){
return message;
}
// setter
public void setMessage(String message) {
this.message = message;
}
}

File: src/main/resources/local_properties.yaml
Comment (PoojaMohite-Khoros) [inline src/main/resources/local_properties.yaml]: excludedCipherSuites what does this do ?

**Review by PoojaMohite-Khoros — CHANGES_REQUESTED** [review]
Please address all reviews and also do not merge to master until approved

**Review by SupriyaPamulapati — APPROVED** [review]
i feel like if u have added all access  keys in properties file it would be still more better 
 
 Good job Raushan ,

**Review by vglalitha — APPROVED** [review]
I have reviewed your code, you have addressed all the changes required.
Good job.

File: src/main/java/com/config/TWConfiguration.java
Comment (ghost) [inline src/main/java/com/config/TWConfiguration.java:5]: Unused imports can be removed here(javax.validation and fasterxml).

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/main/java/com/resource/TwitterWorks.java
Comment (ghost) [inline src/main/java/com/resource/TwitterWorks.java]: Remove unused imports.

File: src/main/java/com/resource/TwitterWorks.java
Comment (ghost) [inline src/main/java/com/resource/TwitterWorks.java]: What is the use of this statement here?(System.out.println(tweet))

**Review by ghost — APPROVED** [review]

File: src/main/java/com/resource/TwitterWorks.java
Comment (raj8789) [reply src/main/java/com/resource/TwitterWorks.java]: ok done

File: src/main/java/com/config/TWConfiguration.java
Comment (raj8789) [reply src/main/java/com/config/TWConfiguration.java:5]: done

File: pom.xml
Comment (raj8789) [reply pom.xml]: removed

File: src/main/java/com/resource/RetrieveTweets.java
Comment (raj8789) [reply src/main/java/com/resource/RetrieveTweets.java]: removed

File: src/main/java/com/resource/SendTweet.java
Comment (raj8789) [reply src/main/java/com/resource/SendTweet.java]: removed

File: src/main/java/com/resource/TwitterWorks.java
Comment (raj8789) [reply src/main/java/com/resource/TwitterWorks.java]: removed

File: src/main/java/com/resource/TwitterWorks.java
Comment (raj8789) [reply src/main/java/com/resource/TwitterWorks.java]: i gave sensible path

File: src/main/java/com/resource/TwitterWorks.java
Comment (raj8789) [reply src/main/java/com/resource/TwitterWorks.java]: done

File: src/main/resources/local_properties.yaml
Comment (raj8789) [reply src/main/resources/local_properties.yaml]: that was added extra i removed

File: src/main/java/com/config/TWConfiguration.java
Comment (raj8789) [reply src/main/java/com/config/TWConfiguration.java:5]: i removed that part

File: src/main/java/com/resource/TwitterWorks.java
Comment (raj8789) [reply src/main/java/com/resource/TwitterWorks.java]: removed used imports

File: src/main/java/com/resource/TwitterWorks.java
Comment (raj8789) [reply src/main/java/com/resource/TwitterWorks.java]: that part is removed

File: src/main/java/com/config/TWConfiguration.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/config/TWConfiguration.java]: rename cb object name

File: src/main/java/com/resource/RetrieveTweets.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/RetrieveTweets.java]: Try to return List<String> instead of Array of String 
https://www.geeksforgeeks.org/list-interface-java-examples/
Please refer above link for examples

File: src/main/java/com/resource/RetrieveTweets.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/RetrieveTweets.java]: status1 is  not a good variable name

File: src/main/java/com/resource/RetrieveTweets.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/RetrieveTweets.java:25]: catch block is empty 
Y is the exception not handled here

File: src/main/java/com/resource/RetrieveTweets.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/RetrieveTweets.java]: use list and also 
 ar[0] = "No Tweet Found on Timeline";
is shady I feel 
use list add/remove methods

Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/RetrieveTweets.java]: also return empty list when no tweets avaiable

File: src/main/java/com/resource/TwitterWorks.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/TwitterWorks.java]: Same review, 
Please do not use String array
We have Collections in JAva that do all the hard work for us
https://www.javatpoint.com/collections-in-java

File: src/main/java/com/resource/TwitterWorks.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/TwitterWorks.java]: Change the class name please

File: src/main/java/com/resource/TwitterWorks.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/TwitterWorks.java]: Add a new file for this, and shift this class to that java file

File: src/main/java/com/twit/TwitterRunner.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/twit/TwitterRunner.java]: Sout please remove

**Review by PoojaMohite-Khoros — CHANGES_REQUESTED** [review]
Please address all code reviews

File: src/main/java/com/config/TWConfiguration.java
Comment (PoojaMohite-Khoros) [reply src/main/java/com/config/TWConfiguration.java:10]: did u get a chance to change object name here, follow sensible variable naming conventions

File: src/main/java/com/resource/RetrieveTweets.java
Comment (raj8789) [reply src/main/java/com/resource/RetrieveTweets.java]: can i add " No tweet Found on Timeline " in Empty List so that user will get a message rather than returning a Empty String

File: src/main/java/com/config/TWConfiguration.java
Comment (raj8789) [reply src/main/java/com/config/TWConfiguration.java]: changed

File: src/main/java/com/resource/RetrieveTweets.java
Comment (raj8789) [reply src/main/java/com/resource/RetrieveTweets.java]: now returning object of ArrayList<String>.

File: src/main/java/com/resource/RetrieveTweets.java
Comment (raj8789) [reply src/main/java/com/resource/RetrieveTweets.java]: now Variable name is status

File: src/main/java/com/resource/RetrieveTweets.java
Comment (raj8789) [reply src/main/java/com/resource/RetrieveTweets.java:25]: added stattement e.printStackTrace();

File: src/main/java/com/resource/RetrieveTweets.java
Comment (raj8789) [reply src/main/java/com/resource/RetrieveTweets.java]: i have added message "No tweet Found on TimeLine " in ArrayList  object if No Tweets on timeline

File: src/main/java/com/resource/TwitterWorks.java
Comment (raj8789) [reply src/main/java/com/resource/TwitterWorks.java]: yes i did by ArrayList object.

File: src/main/java/com/resource/TwitterWorks.java
Comment (raj8789) [reply src/main/java/com/resource/TwitterWorks.java]: i tried to rename the file but not succeed so i created a file SendRetrieveTweets.java and added all TwitterWorks.java content in it.  deleted  TwitterWorks.java

File: src/main/java/com/resource/TwitterWorks.java
Comment (raj8789) [reply src/main/java/com/resource/TwitterWorks.java]: yes i created a new java file TweetPostRequest.java and class TweetPostRequest added in it

File: src/main/java/com/twit/TwitterRunner.java
Comment (raj8789) [reply src/main/java/com/twit/TwitterRunner.java]: yes Sout is removed

File: src/main/java/com/resource/RetrieveTweets.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/RetrieveTweets.java]: rename method name,
fetchLatestTweet

File: src/main/java/com/resource/SendRetrieveTweets.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/SendRetrieveTweets.java]: Confusing class name 
TwitterController

File: src/main/java/com/resource/SendRetrieveTweets.java
Comment (PoojaMohite-Khoros) [inline src/main/java/com/resource/SendRetrieveTweets.java]: rename timeLine
getTweets

**Review by PoojaMohite-Khoros — APPROVED** [review]
Please address minor reviews. Will approve
Merge after addressing reviews.

File: src/main/java/com/resource/RetrieveTweets.java
Comment (raj8789) [reply src/main/java/com/resource/RetrieveTweets.java]: yes i changed the name

File: src/main/java/com/resource/SendRetrieveTweets.java
Comment (raj8789) [reply src/main/java/com/resource/SendRetrieveTweets.java]: name is changed

File: src/main/java/com/resource/SendRetrieveTweets.java
Comment (raj8789) [reply src/main/java/com/resource/SendRetrieveTweets.java]: done

**Pull request merged**
